instrumentation.go

```diff
@@ -452,15 +449,6 @@ func WithTraceExporter(exp trace.SpanExporter) InstrumentationOption {
 	})
 }
 
-// WithSampler returns an [InstrumentationOption] that will configure
-// an [Instrumentation] to use the provided sampler to sample OpenTelemetry traces.
-func WithSampler(sampler trace.Sampler) InstrumentationOption {
-	return fnOpt(func(_ context.Context, c instConfig) (instConfig, error) {
-		c.sampler = sampler
-		return c, nil
-	})
-}
-
```

Comment:
I think there still needs to be programmatic configuration of the samplers that is going to be used. Likely we need to replace the argument here with some enum like we did for logging.